### ISSUE TITLE
Feature Request: Export CSV with full transaction history

### ISSUE THREAD

**Issue by kerooke** [stackernews/stacker.news#217]
As Stacker News sees more and more Bitcoin transaction volume on the platform, it will become important to add in features that make tax reporting easy.

CSV file exports are the preferred format for accountants to keep track of transaction history today, and Stacker News could add a button on a user's wallet page that offers a full download of their transaction history with corresponding fiat values.

For example, a table with the following column names would cover basically everything an accountant would need to know about someone's Stacker News usage:

- Transaction type (boost, tip, reward, funding, withdrawing, etc...)
- Date (and time if possible)
- Sats value spent (when applicable)
- Sats value received (when applicable)
- Fiat value spent (when applicable)
- Fiat value received (when applicable)
- Invoice (for funding/withdrawals)

The fiat values above could be computed by pulling in the preferred fiat currency of Stacker News users as per their profile settings.

**Comment by huumn** [comment]
Re: fiat values

These should be the values that the time of the transaction right? So we'd need some record of spot price history.

**Comment by kristapsk** [comment]
I see this as very useful feature too! (was going to open a feature request issue myself, but then found this)

Fiat value aren't required if there is date/time. You can after export get this yourself, by querying some website, that provides historical BTC/fiat prices.

**Comment by huumn** [comment]
Minimizing the requirement is a good idea - the price part is the hardest requirement really. It's otherwise pretty trivial.

**Comment by kristapsk** [comment]
Yes, I think it's ok to, for at least start, just add CSV export for what is already shown when you look at wallet history.
